frontend/awx/administration/instance-groups/InstanceGroupPage/hooks/useAssociateInstanceModal.tsx

```diff
@@ -0,0 +1,52 @@
+import { useEffect, useState } from 'react';
+import { MultiSelectDialog, usePageDialogs } from '../../../../../../framework';
+import { useAwxView } from '../../../../common/useAwxView';
+import { Instance } from '../../../../interfaces/Instance';
+import { useInstancesColumns } from '../../../instances/hooks/useInstancesColumns';
+import { useInstancesFilters } from '../../../instances/hooks/useInstancesFilter';
+import { awxAPI } from '../../../../common/api/awx-utils';
+import { useTranslation } from 'react-i18next';
+
+export interface AssociateInstanceModalProps {
+  instanceGroupId: string;
+  onAssociate: (items: Instance[]) => void;
+}
+
+function AssociateInstanceModal(props: AssociateInstanceModalProps) {
+  const { t } = useTranslation();
+  const tableColumns = useInstancesColumns();
+  const toolbarFilters = useInstancesFilters();
+  const { instanceGroupId, onAssociate } = props;
+
+  const view = useAwxView<Instance>({
+    url: awxAPI`/instances/`,
+    queryParams: { not__node_type: ['control', 'hop'], not__rampart_groups__id: instanceGroupId },
```

Comment:
```suggestion
    queryParams: { not__node_type: 'control,hop', not__rampart_groups__id: instanceGroupId },
```
I think this change will work and it will prevent you from having to change the queryparams for hub and inside the `useAwxView`.

Reply:
I tried making the query with the comma and it wasn't working. And then I decided to add list support. The query with the comma doesn't filter out control or hop types.